src/http.rs

```diff
@@ -0,0 +1,37 @@
+use anyhow::{Context, Error, Result};
+use reqwest::header::{ACCEPT, AUTHORIZATION, USER_AGENT};
+use reqwest::{Client, Method, RequestBuilder};
+use std::collections::HashMap;
+
+pub(crate) const GH_API_RATE_LIMIT_ENDPOINT: &str = "https://api.github.com/rate_limit";
+
+pub(crate) fn get(token: &str, url: &str) -> RequestBuilder {
+    Client::new()
```

Comment:
Ideally the `Client` should be shared between invocations, to be able to reuse existing connections to GitHub and avoiding the delay of setting up an HTTPS connection.

Reply:
I see,, and how do you envision that implementation, a http client instance in each collector's field or you wanna go "bolder" and go for some sort of global http client for both collectors?

Reply:
I'd have a single client for all the application.

Reply:
like a global static ? or some ClientWrapper that all collectors request through channels or something?

OK,  i'll try to ping you on zulip for further discussion about it and to know more during the next days. 👍

Reply:
Reqwest's clients are really nice because they can be used concurrently and can be freely cloned while still maintaining the benefits of reusing connections. There is no need to use channels thankfully.

I think the best approach is to just have each collector require a `Client` in its constructor and store it in each struct:

```rust
pub async fn new(config: &Config, http: Client) -> Result<Self>;
```

...and then provide it while registering the collectors:

```rust
let http = Client::new();

GitHubRateLimit::new(&p.config, http.clone());
GithubRunners::new(&p.config, http.clone());
// ...
```

Reply:
ok, great! thank you for your time on the clarification and code example! it's clear now, I'll do it!

So I can do this change for the `GithubRunners` only in this PR and defer the refactor of `RateLimit` collector when I'll work in #18 on a future PR or i can do both in this PR. 

How do you prefer me to work on it for you to review?

(PS - I just don't want to run the risk of _overloading_ you in this PR with this change like it happened in https://github.com/rust-lang/monitorbot/pull/13#issuecomment-736584556)

Reply:
I think doing the change just for `GithubRunners` right now is better, I'm a bit busy these days and the less I have to review in one batch the better :)

Reply:
ok, I've pushed the change that add the http client to gh runner's collector. 👍